components/brave_shields/browser/ad_block_default_source_provider.cc

```diff
@@ -0,0 +1,96 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_shields/browser/ad_block_default_source_provider.h"
+
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "base/files/file_path.h"
+#include "base/logging.h"
+#include "base/task/thread_pool.h"
+#include "brave/components/brave_shields/browser/ad_block_component_installer.h"
+#include "content/public/browser/browser_task_traits.h"
+
+#define DAT_FILE "rs-ABPFilterParserData.dat"
+#define REGIONAL_CATALOG "regional_catalog.json"
+const char kAdBlockResourcesFilename[] = "resources.json";
+
+namespace brave_shields {
+
+AdBlockDefaultSourceProvider::AdBlockDefaultSourceProvider(
+    component_updater::ComponentUpdateService* cus,
+    base::RepeatingCallback<void(const std::string& regional_catalog)>
+        regional_catalog_available_cb)
+    : regional_catalog_available_cb_(std::move(regional_catalog_available_cb)) {
+  // Can be nullptr in unit tests
+  if (cus) {
+    RegisterAdBlockDefaultComponent(
+        cus,
+        base::BindRepeating(&AdBlockDefaultSourceProvider::OnComponentReady,
```

Comment:
this causes the SourceProvider to be called on more than one thread because `OnComponentReady` is called on the UI thread and Load is called from the adblock thread